### PR TITLE
Control the depth of the pnpm dependency tree

### PR DESCRIPTION
**Pull request by attiasas**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [ ] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----

## PNPM improvements
* Add option to control the depth of the pnpm dependency tree (using `--max-tree-depth` flag
* Show pnpm install cmd log If error occur

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
       - name: Install dotnet
         uses: actions/setup-dotnet@v3
         with:
-          dotnet-version: '6.x'
```
Comment (hadarshjfrog) [inline .github/workflows/test.yml:54]: relevant here? 
We have a different PR that handles it

File: cli/docs/flags.go [hunk 2]
```diff
@@ -238,17 +240,18 @@ var flagsMap = map[string]components.Flag{
...
+	Npm:          components.NewBoolFlag(Npm, "Set to true to request audit for a npm project."),
+	Pnpm:         components.NewBoolFlag(Pnpm, "Set to true to request audit for a Pnpm project."),
+	Yarn:         components.NewBoolFlag(Yarn, "Set to true to request audit for a Yarn project."),
+	Nuget:        components.NewBoolFlag(Nuget, "Set to true to request audit for a .NET project."),
+	Pip:          components.NewBoolFlag(Pip, "Set to true to request audit for a Pip project."),
+	Pipenv:       components.NewBoolFlag(Pipenv, "Set to true to request audit for a Pipenv project."),
+	Poetry:       components.NewBoolFlag(Poetry, "Set to true to request audit for a Poetry project."),
+	Go:           components.NewBoolFlag(Go, "Set to true to request audit for a Go project."),
+	DepType:      components.NewStringFlag(DepType, "[npm] Defines npm dependencies type. Possible values are: all, devOnly and prodOnly."),
+	MaxTreeDepth: components.NewStringFlag(MaxTreeDepth, "[pnpm] Max depth of the generated dependencies tree for SCA scan.", components.WithStrDefaultValue("Infinity")),
```
Comment (hadarshjfrog) [inline cli/docs/flags.go:254]: Is it only pnpm? 
isn't it relevant for npm? other technologies?

If it's a flag specific for a technologies - I'd add its name to the flag, like `PnpmMaxTreeDepth `  (same with `DepType`)

**Review by hadarshjfrog — COMMENTED** [review]
flag name + dotnet version

File: cli/docs/flags.go (hunk 2 above)
Comment (attiasas) [reply cli/docs/flags.go:256]: for now it's specific for `pnpm`, we can extend its usage later, changing the description is easier than changing the flag itself.
`DepType` could also be used for multiple techs

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -47,7 +47,7 @@ jobs:
       - name: Install NuGet
         uses: nuget/setup-nuget@v2
         with:
-          nuget-version: 6.11.0
+          nuget-version: 6.x
```
Comment (hadarshjfrog) [inline .github/workflows/test.yml:50]: Do we still need it? 
This what caused the doCurationAudit in Windows tests?

File: cli/docs/flags.go (hunk 2 above)
Comment (hadarshjfrog) [reply cli/docs/flags.go:256]: I think it's relevant at least of npm as well...

File: commands/audit/sca/pnpm/pnpm.go [hunk 4]
```diff
@@ -120,13 +120,16 @@ func installProjectIfNeeded(pnpmExecPath, workingDir string) (dirForDependencies
...
-	err = getPnpmCmd(pnpmExecPath, dirForDependenciesCalculation, "install", npm.IgnoreScriptsFlag).GetCmd().Run()
+	output, err := getPnpmCmd(pnpmExecPath, dirForDependenciesCalculation, "install", npm.IgnoreScriptsFlag).GetCmd().CombinedOutput()
+	if err != nil {
+		err = fmt.Errorf("failed to install project: %w\n%s", err, string(output))
+	}
 	return
 }
 
 // Run 'pnpm ls ...' command (project must be installed) and parse the returned result to create a dependencies trees for the projects.
 func calculateDependencies(executablePath, workingDir string, params utils.AuditParams) (dependencyTrees []*xrayUtils.GraphNode, uniqueDeps []string, err error) {
```
Comment (hadarshjfrog) [inline commands/audit/sca/pnpm/pnpm.go:131]: I'd like to add a test case here - in  `TestBuildDependencyTree` in  `pnpm_test.go` where we pass once - and see we get a limited tree - and without it - to see that we get "Infinity"

**Review by hadarshjfrog — COMMENTED** [review]
a question, suggestion, and test addition request

File: cli/docs/flags.go (hunk 2 above)
Comment (attiasas) [reply cli/docs/flags.go:256]: we need to implement it to be relevant, the only implementation is for pnpm. we can expend this flag if needed, right now it only affects pnpm

File: commands/audit/sca/pnpm/pnpm.go (hunk 4 above)
Comment (attiasas) [reply commands/audit/sca/pnpm/pnpm.go:132]: done

File: commands/audit/sca/pnpm/pnpm_test.go [hunk 5]
```diff
@@ -16,6 +17,67 @@ import (
...
+	// Create and change directory to test workspace
+	_, cleanUp := sca.CreateTestWorkspace(t, filepath.Join("projects", "package-managers", "npm", "npm-big-tree"))
+	defer cleanUp()
+	testCases := []struct {
+		name               string
+		treeDepth          string
+		expectedUniqueDeps []string
+		expectedTree       *xrayUtils.GraphNode
+	}{
+		// {
```
Comment (hadarshjfrog) [inline commands/audit/sca/pnpm/pnpm_test.go:30]: can we remove dead code?

**Review by hadarshjfrog — APPROVED** [review]
Please remove dead code and merge

**Pull request merged**